doc/sphinx/Pacemaker_Explained/utilization.rst

```diff
@@ -98,167 +86,140 @@ tool can be used to set these attributes.
+Order of Resource Assignment
+############################
 
-* The resource that has the highest ``priority`` (see :ref:`resource_options`) gets
-  assigned first.
+When assigning resources to nodes, the cluster chooses the next one to assign
+by considering the following criteria one by one until a single resource is
+selected:
 
-* If their priorities are equal, check whether they are already running. The
-  resource that has the highest score on the node where it's running gets assigned
-  first, to prevent resource shuffling.
+* Assign the resource with the highest :ref:`priority <meta_priority>`.
 
-* If the scores above are equal or the resources are not running, the resource has
-  the highest score on the preferred node gets assigned first.
+* If any resources are already active, assign the one with the highest score on
+  its current node. This avoids unnecessary resource shuffling.
 
-* If the scores above are equal, the first runnable resource listed in the CIB
-  gets assigned first.
+* Assign the resource with the highest score on its preferred node.
 
-Limitations and Workarounds
-###########################
+* If more than one resource remains after considering all other criteria,
+  assign the one of them that is listed first in the CIB.
+
+.. note::
+
+   For bundles, only the priority set for the bundle itself matters. If the
+   bundle contains a primitive, the primitive's priority is ignored.
+
+Limitations
+###########
 
 The type of problem Pacemaker is dealing with here is known as the
-`knapsack problem <http://en.wikipedia.org/wiki/Knapsack_problem>`_ and falls into
-the `NP-complete <http://en.wikipedia.org/wiki/NP-complete>`_ category of computer
-science problems -- a fancy way of saying "it takes a really long time
-to solve".
+`knapsack problem <https://en.wikipedia.org/wiki/Knapsack_problem>`_ and falls
+into the `NP-complete <https://en.wikipedia.org/wiki/NP-completeness>`_
+category of computer science problems -- a fancy way of saying "it takes a
+really long time to solve".
 
-Clearly in a HA cluster, it's not acceptable to spend minutes, let alone hours
-or days, finding an optimal solution while services remain unavailable.
+In a high-availability cluster, it is unacceptable to spend minutes, let alone
+hours or days, finding an optimal solution while services are down.
 
-So instead of trying to solve the problem completely, Pacemaker uses a
-*best effort* algorithm for determining which node should host a particular
-service. This means it arrives at a solution much faster than traditional
-linear programming algorithms, but by doing so at the price of leaving some
-services stopped.
+Instead of trying to solve the problem completely, Pacemaker uses a "best
+effort" algorithm. This arrives at a quick solution, but at the cost of
+possibly leaving some resources stopped unnecessarily.
 
-In the contrived example at the start of this chapter:
+Using the example configuration at the start of this chapter:
```

Comment:
May want to specify `balanced` placement strategy (at least that's my assumption)

Reply:
thanks